integration/build.gradle

```diff
@@ -14,38 +14,40 @@ project.configurations {
 }
 
 dependencies {
-    compile project(':hermes-common')
-    compile project(':hermes-consumers')
-    compile project(':hermes-frontend')
-    compile project(':hermes-management')
-    compile project(':hermes-client')
-    compile project(':hermes-tracker')
-
-    testCompile project(':hermes-test-helper')
-    testCompile group: 'org.apache.commons', name: 'commons-lang3', version: '3.0'
-    testCompile group: 'commons-beanutils', name: 'commons-beanutils', version: '1.9.4'
-    testCompile group: 'org.apache.curator', name: 'curator-test', version: versions.curator
-    testCompile group: 'org.springframework', name: 'spring-webflux', version: versions.spring_web
-    testCompile group: 'com.squareup.okhttp3', name: 'okhttp', version: versions.okhttp
-
-    testCompile group: 'org.testcontainers', name: 'testcontainers', version: versions.testcontainers
-    testCompile group: 'org.testcontainers', name: 'gcloud', version: versions.testcontainers
-
-    testCompile files('./lib/schema-repo-server-0.1.3-jersey2.jar');
-    testCompile group: 'org.glassfish.jersey.containers', name: 'jersey-container-grizzly2-http', version: versions.jersey
-    testCompile group: 'org.freemarker', name: 'freemarker', version: '2.3.21'
-    testCompile group: 'org.schemarepo', name: 'schema-repo-common', version: '0.1.3'
-    testCompile group: 'org.eclipse.jetty', name: 'jetty-alpn-openjdk8-server', version: '9.4.28.v20200408'
-    testCompile group: 'org.eclipse.jetty', name: 'jetty-alpn-openjdk8-client', version: '9.4.30.v20200611'
-    testCompile group: 'org.eclipse.jetty', name: 'jetty-servlet', version: versions.jetty
-    testCompile group: 'org.eclipse.jetty', name: 'jetty-servlets', version: versions.jetty
-    testCompile group: 'org.glassfish.jersey.bundles.repackaged', name: 'jersey-guava', version: '2.23.2'
-
-    testCompile(group: 'org.hornetq', name: 'hornetq-jms-server', version: '2.4.1.Final') {
+    implementation project(':hermes-common')
+    implementation project(':hermes-consumers')
+    implementation project(':hermes-frontend')
+    implementation project(':hermes-management')
+    implementation project(':hermes-client')
+    implementation project(':hermes-tracker')
+
+    implementation project(':hermes-test-helper')
+    //TODO: why testImplementations do not work?
```

Comment:
**note:** Great question which I'm more than glad to answer :)

Module `integration` uses [Java Plugin](https://docs.gradle.org/current/userguide/java_plugin.html), which gives you `implementation` [configuration](https://docs.gradle.org/current/userguide/declaring_dependencies.html#sec:what-are-dependency-configurations). To define our custom configuration - `integration` - we use configuration extending `implementation`.

Here you can find definition of `integration` configuration:
```Gradle
project.configurations {
    integration {
        extendsFrom project.configurations.runtimeClasspath
        description = 'Dependencies for integration tests'
        transitive = true
        visible = true
    }
    alpnboot
}
```

Crucial thing here is `extendsFrom project.configurations.runtimeClasspath` line. 
`runtimeClasspath` is (yet another) configuration which [extends `implementation` configuration](https://docs.gradle.org/current/userguide/java_plugin.html#tab:configurations):

> **runtimeClasspath extends** runtimeOnly, **implementation**
>   Runtime classpath contains elements of the implementation, as well as runtime only elements.

So whole configuration chain looks like this:
```
+--------------+      +----------------+     +-----------+
|implementation|<-----+runtimeClasspath|<----+integration|
+--------------+      +----------------+     +-----------+
              ^
              |
              |       +------------------+
              +-------+testImplementation|
                      +------------------+
```

And as you can see, `integration` configuration does not use `testImplementation` configuration resources - that's why using `testImplementation` doesn't work.

___

PS - using `integration` instead of `implementation` would also do the job 😸

Reply:
great explanation, thank you!